bin/node-template/node/src/command.rs

```diff
@@ -66,15 +66,55 @@ impl SubstrateCli for Cli {
 pub fn run() -> sc_cli::Result<()> {
 	let cli = Cli::from_args();
 
-	match cli.subcommand {
-		Some(ref subcommand) => {
-			let runner = cli.create_runner(subcommand)?;
-			runner.run_subcommand(subcommand, |config| {
-				let PartialComponents { client, backend, task_manager, import_queue, .. }
+	match &cli.subcommand {
+		Some(Subcommand::BuildSpec(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			runner.sync_run(|config| cmd.run(config.chain_spec, config.network))
```

Comment:
```suggestion
			let runner = cli.create_runner(cmd)?;
			runner.sync_run(|config| cmd.run(config.chain_spec, config.network))
```

Can this not be implemented as method the command?

```
fn run(cli: &Cli, chain_spec: &Box<ChainSpec>, network: &Network) {}
```

And this for all subcommands? Otherwise we always need to copy this code here.

Reply:
I'm not sure what you mean.

Reply:
Instead of writing every time:
```
let runner = cli.create_runner(cmd)?;
			runner.async_run(|config| {
				let PartialComponents { client, task_manager, ..}
					= new_partial(&config)?;
				Ok((cmd.run(client, config.database), task_manager))
			})
```

There is a function with a name `run` implemented for each subcommand and you call it like this:
```
cmd.run(cli, |config| { new_partial(config).map(|v| (v.client, v.task_manager) })
```

This function is doing internally the same, as the code that is currently written. (The one I pasted as first)

Reply:
This works well for substrate, but not for polkadot, where you need to know which runtime to use for a function like:
```rust
cmd.run(&cli, |config| {
    let (client, _, import_queue, task_manager) = service::new_chain_ops::<
        service::kusama_runtime::Runtime, service::KusamaExecutor,
    >(config)?;
    Ok((client, import_queue, task_manager))
}
```

The only way to do this is to get `chain_spec.is_kusama() ` is to create a runner first, which isn't do-able in this case. So I think the best option would be to revert this change.

Reply:
Hmm yeah.